src/metabase/api/table.clj

```diff
@@ -78,6 +68,34 @@
         (sync/sync-table! updated-table))
       updated-table)))
 
+(api/defendpoint PUT "/:id"
+  "Update `Table` with ID."
+  [id :as {{:keys [display_name entity_type visibility_type description caveats points_of_interest
+                   show_in_getting_started], :as body} :body}]
+  {display_name            (s/maybe su/NonBlankString)
+   entity_type             (s/maybe su/EntityTypeKeywordOrString)
+   visibility_type         (s/maybe TableVisibilityType)
+   description             (s/maybe su/NonBlankString)
+   caveats                 (s/maybe su/NonBlankString)
+   points_of_interest      (s/maybe su/NonBlankString)
+   show_in_getting_started (s/maybe s/Bool)}
+  (update-table id body))
+
+(api/defendpoint PUT "/bulk"
+  "Update all `Table` in `ids`."
+  [:as {{:keys [ids display_name entity_type visibility_type description caveats points_of_interest
+            show_in_getting_started], :as body} :body}]
+  {ids                     (s/pred vector?)
```

Comment:
you should do `(su/non-empty [su/IntGreaterThanZero])` instead of `(s/pred vector?)` -- that will check that it's sequential (e.g. a vector or a list); that every element is an integer greater than zero (i.e. a valid ID), and that the sequence isn't empty

Those stuff in `su` also generates API documentation vs a random unnamed schema

Reply:
```clj
(su/api-error-message (su/non-empty [su/IntGreaterThanZero]))
;; -> "value must be an array. Each value must be an integer greater than zero. The array cannot be empty."

(su/api-error-message (s/pred vector?))
;; -> nil
```